### PR TITLE
[android] Add explicit check for PointChooserMode

### LINKED ISSUE #3945: [android] Crash in the Editor
```
--------- beginning of crash
11-22 13:16:57.331 E/AndroidRuntime( 7260): FATAL EXCEPTION: main
11-22 13:16:57.331 E/AndroidRuntime( 7260): Process: app.organicmaps, PID: 7260
11-22 13:16:57.331 E/AndroidRuntime( 7260): java.lang.IllegalStateException: Unexpected mPositionChooserMode
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at app.organicmaps.MwmActivity.lambda$initPositionChooser$3(Unknown Source:24)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at app.organicmaps.MwmActivity.i(Unknown Source:0)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at app.organicmaps.x.onClick(Unknown Source:2)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.view.View.performClick(View.java:7862)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.widget.TextView.performClick(TextView.java:15004)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.view.View.performClickInternal(View.java:7831)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.view.View.access$3600(View.java:879)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.view.View$PerformClick.run(View.java:29359)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.os.Handler.handleCallback(Handler.java:883)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.os.Handler.dispatchMessage(Handler.java:100)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.os.Looper.loop(Looper.java:237)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at android.app.ActivityThread.main(ActivityThread.java:8167)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at java.lang.reflect.Method.invoke(Native Method)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
11-22 13:16:57.331 E/AndroidRuntime( 7260): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
--------- beginning of main
```

### PR DESCRIPTION
**Pull request by rtsisyk**
It is not clear how to reproduce this bug. Add a workaround and a log entry.

See #3945
See #3593

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/MwmActivity.java [hunk 1]
```diff
@@ -1034,6 +1034,12 @@ protected void onResume()
     {
       UiUtils.show(mPointChooser);
       mMapButtonsViewModel.setButtonsHidden(true);
+      if (mPointChooserMode == PointChooserMode.NONE)
+      {
+        // BUG: https://github.com/organicmaps/organicmaps/issues/3945
+        Logger.e(TAG, "nativeIsInChoosePositionMode is true but mPointChooserMode is NONE");
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/MwmActivity.java:1040]: Let's explicitly crash here to find and fix the bug reason.

Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/MwmActivity.java]: It is pretty obvious that in #3945 the app started with `Framework.nativeIsInChoosePositionMode() &&  mPointChooserMode == PointChooserMode.NONE`. I don't know why. This state is managed somewhere in the native editor.

**Review by biodranik — APPROVED** [review]
Is it the best way to debug it?

**Comment by rtsisyk** [comment]
> Is it the best way to debug it?

Let's try to reproduce first.

**Pull request merged**